p2panda-rs/src/storage_provider/models.rs

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+//! Data models which are used in StorageProvider.
+
+use std::fmt::Debug;
+
+use super::StorageProviderError;
+use crate::document::DocumentId;
+use crate::entry::{decode_entry, EntrySigned, LogId};
+use crate::identity::Author;
+use crate::operation::OperationEncoded;
+use crate::schema::SchemaId;
+use crate::Validate;
+
+/// Struct wrapping an entry with it's operation.
+///
+/// Used internally throughout `storage_provider` in method args and default trait definitions.
+/// The `AsStorageEntry` trait requires `TryFrom<EntryWithOperation>` & `TryInto<EntryWithOperation>`
+/// conversion traits to be present.
```

Comment:
I was trying to emphasise that this is a type which is only used internally within the `storage_provider` rather than a general use p2panda type. Maybe that's obvious?

Reply:
Slightly off topic, but does the pattern between in memory and db types make sense to you? 

`EntryWithOperation` <-> `StorageEntry` & `Log` <-> `StorageLog`

Reply:
> I was trying to emphasise that this is a type which is only used internally within the `storage_provider` rather than a general use p2panda type. Maybe that's obvious?

Its a relatively useful struct, how about moving it into `entry` or `operations` module?

Related: https://github.com/p2panda/p2panda/issues/198

Reply:
> Slightly off topic, but does the pattern between in memory and db types make sense to you?

Yes! It's great!